### ISSUE TITLE
`latest` image ensure does not refetch img from remote registry

### ISSUE THREAD

**Issue by dmnfortytwo** [puppetlabs/puppetlabs-docker#348]
Puppet does not refetch image even with 'latest' ensure. 

```
  docker::image { 'artifactory.foo.bar:5000/proj/deploy':
    ensure    => latest,
    image_tag => 'latest'
  }
```

```
root@deploytest-gce-be-2:~# puppet agent --enable; puppet agent -t; puppet agent --disable
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for deploytest-gce-be-2.foo.bar
Info: Unable to serialize catalog to json, retrying with pson
Info: Applying configuration version '1538503453'
Notice: Applied catalog in 19.55 seconds
root@deploytest-gce-be-2:~# docker images
REPOSITORY                                    TAG                 IMAGE ID            CREATED             SIZE
artifactory.foo.bar:5000/proj/deploy   latest              1cee6edd911d        35 minutes ago      652MB
artifactory.foo.bar:5000/proj/deploy   v15                 e1753b369545        25 hours ago        652MB
ubuntu                                        18.04               cd6d8154f1e1        3 weeks ago         84.1MB
ubuntu                                        latest              cd6d8154f1e1        3 weeks ago         84.1MB
root@deploytest-gce-be-2:~# docker pull artifactory.foo.bar:5000/proj/deploy:latest
latest: Pulling from proj/deploy
124c757242f8: Already exists 
9d866f8bde2a: Already exists 
fa3f2f277e67: Already exists 
398d32b153e8: Already exists 
afde35469481: Already exists 
bbcd32f40191: Pull complete 
e1cd26d0d757: Pull complete 
250b3431592f: Pull complete 
39b1619d5877: Pull complete 
281c47298ce5: Pull complete 
f0ffc76575dd: Pull complete 
426512d98627: Pull complete 
9a0b68dd8173: Pull complete 
Digest: sha256:c2aa261667ce63e44f953e89ed2c98829fb2bf0d9591ce34df4f1669931ff01f
Status: Downloaded newer image for artifactory.foo.bar:5000/proj/deploy:latest
root@deploytest-gce-be-2:~# docker images
REPOSITORY                                    TAG                 IMAGE ID            CREATED             SIZE
artifactory.foo.bar:5000/proj/deploy   latest              c05a8bd124f9        11 minutes ago      652MB
artifactory.foo.bar:5000/proj/deploy   <none>              1cee6edd911d        36 minutes ago      652MB
artifactory.foo.bar:5000/proj/deploy   v15                 e1753b369545        25 hours ago        652MB
ubuntu                                        18.04               cd6d8154f1e1        3 weeks ago         84.1MB
ubuntu                                        latest              cd6d8154f1e1        3 weeks ago         84.1MB

```

```
root@deploytest-gce-be-2:~# puppet --version
5.5.1
root@deploytest-gce-be-2:~# docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false
root@deploytest-gce-be-2:~# facter os
{
  architecture => "amd64",
  distro => {
    codename => "bionic",
    description => "Ubuntu 18.04.1 LTS",
    id => "Ubuntu",
    release => {
      full => "18.04",
      major => "18.04"
    }
  },
  family => "Debian",
  hardware => "x86_64",
  name => "Ubuntu",
  release => {
    full => "18.04",
    major => "18.04"
  },
  selinux => {
    enabled => false
  }
}

/etc/puppetlabs/code/environments/production/modules
├── config (???)
├── puppet-letsencrypt (v2.0.1)
├── puppet-nginx (v0.11.0)
├── puppetlabs-apt (v4.5.1)  invalid
├── puppetlabs-concat (v4.2.1)
├── puppetlabs-docker (v3.0.0)
├── puppetlabs-inifile (v1.2.0)  invalid
├── puppetlabs-lvm (v1.0.1)
├── puppetlabs-stdlib (v4.25.1)
├── puppetlabs-vcsrepo (v1.2.0)  invalid
├── thias-sysctl (v1.0.6)
```

**Comment by davejrt** [comment]
Due to the way the images are pulled, this functionality was removed in favor of the script not being run during a no-op. There is a note about this change in the README under [containers](https://github.com/puppetlabs/puppetlabs-docker/blob/master/README.md#containers) hence we made this a major release as it is a breaking change. 

The decision was made internally to go with this, as using the latest tag is a bad practice, in any environment let alone production.
